types/sap__approuter/index.d.ts

```diff
@@ -60,23 +60,33 @@ declare namespace approuter {
          */
         backendTimeout?: (req: IncomingMessage, done: () => void) => void;
 
-        /**
-         * Callback for {@link afterRequestHandler}
-         *
-         * @param error - Error object in case of error
-         * @param incomingResponse - Modified incomingResponse
-         */
+        // In order to make middleware easier to write to be agnostic of the route, the req.url will be altered to remove the route part and the original will be available set to this value
+        originalUrl?: string;
+
+        // On each request, the application router executes registered middlewares in a certain order and the session is not available to all of them.
```

Comment:
```suggestion
        /** Note: On each request, the application router executes registered middlewares in a certain order and the session is not available to all of them. */
```
Also JSDoc and some wording.